azure/test.yml

```diff
@@ -24,33 +25,38 @@ jobs:
           vmImage: macOS-10.14
         Windows:
           vmImage: windows-2019
-  ${{ if ne('true', parameters.cross) }}:
-    variables:
+  variables:
+    ${{ if ne('true', parameters.cross) }}:
       vmImage: ubuntu-16.04
+    options: ''
   pool:
     vmImage: $(vmImage)
   steps:
   - template: _setup.yml
     parameters:
       rust: ${{ parameters.rust }}
       setup: ${{ parameters.setup }}
-  - ${{ if ne('true', parameters.single_threaded) }}:
-    - script: cargo test --all --features "${{ parameters.features }}"
-      displayName: Run tests
-      env:
-        ${{ insert }}: ${{ parameters.envs }}
-  - ${{ if eq('true', parameters.single_threaded) }}:
-    - script: cargo test --all --features "${{ parameters.features }}" -- --test-threads=1
-      displayName: Run tests (single-threaded)
+  # Cargo options go here
+  - script: echo '##vso[task.setvariable variable=options]$(options) --all'
+  - ${{ if eq(true, parameters.features)}}:
+    - script: echo '##vso[task.setvariable variable=options]$(options) --features "${{ parameters.features }}"'
+  # Add any additional conditional libtest options to this check
+  - ${{ if or(eq(true, parameters.single_threaded), eq(true, parameters.test_ignored))}}:
+    - script: echo '##vso[task.setvariable variable=options]$(options) --'
+  # Options passed to libtest go here
+  - ${{ if eq(true, parameters.single_threaded) }}:
+    - script: echo '##vso[task.setvariable variable=options]$(options) --test-threads=1'
+  - ${{ if eq(true, parameters.test_ignored)}}:
+    - script: echo '##vso[task.setvariable variable=options]$(options) --ignored'
+  # Run tests with the given options
+  - ${{ if eq(true, parameters.raw_opts) }}:
+    - bash: cargo test ${{ parameters.raw_opts }}
```

Comment:
I was surprised when I saw this -- I would have expected `raw_opts` to be a way to pass _additional_ arguments, not to replace any arguments caused by other options. That would mean that this:
```yaml
parameters:
  single_threaded: true
  raw_opts: "--foobar"
```
does not actually run the tests single-threaded.

Reply:
As an aside to this, what kind of additional parameters are you expecting users to use here?

Reply:
I personally need to use -Z options, which change occasionally.

It was intended that `raw` overrides any other options passed, so as to avoid any weird edge cases down the line. I modeled it off the lxd/ceph raw commands which also override any other args passed

Reply:
Hmm, I worry that that'll be too confusing to users. Maybe if we had a way to emit warnings for particular parameter combinations, but we don't really... I think my inclination here would bet o have `raw_opts` default to `--`, and then inject it at:
```console
$ cargo test --all <other cargo flags> $RAW_OPTS <other libtest flags>
```
That way, users can give both cargo and libtest flags in raw_opts. We'd have to document that they _must_ include the `--` in their raw flags though.

As for the combination of raw options with other options, I would be inclined here to just punt on the issue. That is, include raw opts + the options as dictated by parameters. Usually that will be fine, because the user generally controls those parameters, and can just avoid adding additional options.

Reply:
Okay I'll change it to align with this